packages/api-client/src/index.ts

```diff
@@ -0,0 +1,215 @@
+import { stringify } from 'query-string';
+
+import type { Serialized } from '../../core-typings/dist';
+import type { MatchPathPattern, OperationParams, OperationResult, PathFor } from '../../rest-typings/dist';
+import type { Middleware, RestClientInterface } from './RestClientInterface';
+
+export { RestClientInterface };
+
+const pipe =
+	<T extends (...args: any[]) => any>(fn: T) =>
+	(...args: Parameters<T>): ReturnType<T> =>
+		fn(...args);
+
+function buildFormData(data?: Record<string, any> | void, formData = new FormData(), parentKey?: string): FormData {
+	if (data instanceof FormData) {
+		return data;
+	}
+	if (!data) {
+		return formData;
+	}
+
+	if (typeof data === 'object' && !(data instanceof File)) {
+		Object.keys(data).forEach((key) => {
+			buildFormData(formData, data[key], parentKey ? `${parentKey}[${key}]` : key);
+		});
+	} else {
+		data && parentKey && formData.append(parentKey, data);
+	}
+	return formData;
+}
+
+const checkIfIsFormData = (data: any = {}): boolean => {
+	if (data instanceof FormData) {
+		return true;
+	}
+	return Object.values(data).some((value) => {
+		if (typeof value === 'object' && !(value instanceof File)) {
+			return checkIfIsFormData(value);
+		}
+		return value instanceof File;
+	});
+};
+
+export class RestClient implements RestClientInterface {
+	private readonly baseUrl: string;
+
+	private headers: Record<string, string> = {};
+
+	private credentials:
+		| {
+				'X-User-Id': string;
+				'X-Auth-Token': string;
+		  }
+		| undefined;
+
+	constructor({
+		baseUrl,
+		credentials,
+		headers = {},
+	}: {
+		baseUrl: string;
+		credentials?: {
+			'X-User-Id': string;
+			'X-Auth-Token': string;
+		};
+		headers?: Record<string, string>;
+	}) {
+		this.baseUrl = `${baseUrl}/api`;
+		this.setCredentials(credentials);
+		this.headers = headers;
+	}
+
+	getCredentials(): ReturnType<RestClientInterface['getCredentials']> {
+		return this.credentials;
+	}
+
+	setCredentials: RestClientInterface['setCredentials'] = (credentials) => {
+		this.credentials = credentials;
+	};
+
+	get<TPath extends PathFor<'GET'>>(
+		endpoint: TPath,
+		params: void extends OperationParams<'GET', MatchPathPattern<TPath>> ? never : OperationParams<'GET', MatchPathPattern<TPath>>,
+		options?: Omit<RequestInit, 'method'>,
+	): Promise<Serialized<OperationResult<'GET', MatchPathPattern<TPath>>>>;
+
+	get<TPath extends PathFor<'GET'>>(
+		endpoint: TPath,
+		params?: void extends OperationParams<'GET', MatchPathPattern<TPath>> ? undefined : never,
+		options?: Omit<RequestInit, 'method'>,
+	): Promise<Serialized<OperationResult<'GET', MatchPathPattern<TPath>>>>;
+
+	get<TPath extends PathFor<'GET'>>(
+		endpoint: TPath,
+		params?: OperationParams<'GET', MatchPathPattern<TPath>>,
+		options?: Omit<RequestInit, 'method'>,
+	): Promise<Serialized<OperationResult<'GET', MatchPathPattern<TPath>>>> {
+		if (/\?/.test(endpoint)) {
+			// throw new Error('Endpoint cannot contain query string');
```

Comment:
```suggestion
			// throw new Error('Endpoint cannot contain query string');
```

Why not throw it?

Reply:
I was thinking about first give a friendly reminder for our omni friends and then break every single call